core/objects.go

```diff
@@ -252,6 +265,43 @@ type Challenge struct {
 
 	// Used by dns and dvsni challenges
 	Validation *jose.JsonWebSignature `json:"validation,omitempty"`
+
+	// Contains information about URLs used or redirected to and IPs resolved and
+	// used
+	ValidationRecord []ValidationRecord `json:"validationRecord,omitempty"`
+}
+
+// RecordsSane checks the sanity of a ValidationRecord object before sending it
+// back to the RA to be stored.
+func (ch Challenge) RecordsSane() bool {
```

Comment:
It would be better to just fold this in with `Challenge.IsSane()`, since that already has the branch.  You can also collapse some of the repetition.

```
urlRequired := (ch.Type == ChallengeTypeSimpleHTTP)
if ch.Type != ChallengeTypeSimpleHTTP && len(ch.ValidationRecord) != 1 {
  return false
}
for _, record := range ch.ValidationRecord {
  if ... || (urlRequred && rec.URL == "") {
    return false
  }
}
```